core/Tracker/Request.php

```diff
@@ -522,8 +524,11 @@ public function getIdSite()
          */
         Piwik::postEvent('Tracker.Request.getIdSite', array(&$idSite, $this->params));
 
-        if ($idSite <= 0) {
+        if ($idSite < 0) {
             throw new UnexpectedWebsiteFoundException('Invalid idSite: \'' . $idSite . '\'');
+            // we allow 0... the request will fail anyway as the site won't exist... allowing 0 will help us
+            // reporting this tracking problem as it is a common issue. Otherwise we would not be able to report
+            // this problem in tracking failures
```

Comment:
Does this mean the exception cannot be caught later? If so, could log the failure there.

Reply:
Yes it would otherwise not be caught later. and logging wouldn't work as people wouldn't see the logged error. Like nobody would really look there and eg on cloud people wouldn't even be able to see the logs. We have heaps of logs like `idsite=0` already and want to make users aware of this problem with this PR. It's really the most common tracking error.

Reply:
I meant log through `Failures::logFailure()`, like catch in `Tracker::main()` and log that failure. But if it can't be caught then that's not an option I guess.